### ISSUE TITLE
Stream dynamodb data to a long-lived data bucket in admin for Redshift import

### ISSUE THREAD

**Issue by rlister** [ArcadiaPower/goldfinch#23]
@mjrich Needs access to goldfinch data in Redshift (for queries in `grebe` project).

We _could_ build a "pull" model, where Redshift does `COPY` from goldfinch's dynamodb tables.

Pros:
- easy enough to construct `COPY` query and is a standard Redshift pattern

Cons:
- Redshift in admin account would need creds and permissions to read tables in prod account
- we would need to know tables names (which are not set and may be changed by Cloudformation)
- schema-dependent
- it breaks application isolation: goldfinch is now responsible for grebe data, and grebe needs internal knowledge and permissions for goldfinch

Alternatively, I propose a "push" model, where a new stack in goldfinch streams dynamodb data to a well-known and long-lived S3 bucket in admin/data account. Redshift can then come and fetch data from S3, and it is the responsibility of goldfinch to ensure it gets there.

Pros:
- grebe/Redshift need no knowledge of goldfinch: they just go and look for S3 objects like `goldfinch/master/$date/$tablename/*`
- we only need one cross-account permission: admin allows S3 incoming from prod/stg/dev accounts, as is already set up in hamerkop `s3policy` stack for macaw buckets
- this model is easily extensible to future projects, which can perform a similar task, depending on their internal knowledge of their data store, with the S3 bucket presenting one consistent target for the data team Redshift hackers to work with

Proposal:
- new long-lived data bucket `arcadia-data-dumps` or similar, in admin account
- hamerkop `s3policy` addition to allow prod/dev/stg accounts write on this bucket
- new goldfinch stack `data` or `analytics`, which creates dynamodb streams, and possibly lambdas, to stream dynamodb data as CSV or jsonl (decide with Max) to above bucket
- monitoring for streams and lambdas

**Comment by jjensenmike** [comment]
The push model proposal sounds great. Agree that this pattern is needed and reusable in so many places.

Already mentioned in here and in Slack but Redshift doesn't really play nice with JSON and JSON -> Redshift imports can be a pain. A lot of content to a JSON column that is uploaded into Redshift from a CSV is workable since Redshift deal with that, but not very cleanly.

**Comment by jjensenmike** [comment]
After Nathan mentioned Glue, I see in the documentation that it can be used to easily flatten JSON data. So cool.

**Comment by rlister** [comment]
Yeah, I'll probably initially just do the data import to the data lake, and we can iterate on formats and flattening as a second step.

**Comment by mjrich** [comment]
In case this helps at all for requirements, all I'd need are the daily production amounts per project.  So like `date,project,kwh` : `2018-01-28,Holygrove,3369`. Here's an example screenshot from the Locus system for RG&E for example.

[Link to example screenshot on Slack](https://files.slack.com/files-pri/T02FR7QQD-FFT710W01/screen_shot_2019-01-30_at_5.07.03_pm.png)
